gpbft/gpbft.go

```diff
@@ -288,6 +288,20 @@ func (i *instance) receiveOne(msg *GMessage) error {
 		return nil
 	}
 
+	// Drop message that:
+	// * belong to future rounds, beyond the configured max lookahead threshold, and
+	// * carry no justification, i.e. are spammable.
+	//
+	// The only messages that are spammable are COMMIT for bottom. QUALITY and
+	// PREPARE messages may also not carry justification, but they are not spammable.
+	// Because, such messages are only valid at round 0; therefore, cannot belong
+	// to future rounds.
```

Comment:
This is slightly misleading. PREPARE messages are valid at any round, but only those in round 0 are valid without justification. PREPARE for subsequent rounds must be justified.

Reply:
Oh i see the "vaild" wording; I will adjust thank you. What I meant was that the will not pass `validateMessage` without justification for non-zero rounds.